src/dbsql.py

```diff
@@ -87,3 +91,96 @@ def build_parameter(params: dict) -> Parameter:
+        :param warehouse_id: optional ID of the SQL warehouse to refresh query
+        """
+
+        target_folder_path = "folders/{0}".format(self._get_path_object_id(target_folder))
+
+        # First migrate the query
+        query_id = self.create_query(alert.query, target_folder_path)
+        result = self._create_alert_api_call(query_id, alert, target_folder_path)
+        if alert.schedule and destination_id and warehouse_id:
+            self._create_alert_schedule_api_call(result.id, alert.schedule, destination_id, warehouse_id)
+        return result.id
+
+    def _create_alert_api_call(self, query_id: str, alert: Alert, parent_folder: str):
+        """
+        Creates an alert in Databricks
+        """
+        return self.client.alerts.create(
+            name=alert.name,
+            options=AlertOptions.from_dict(alert.options),
+            query_id=query_id,
+            parent=parent_folder,
+            rearm=alert.rearm,
+        )
+
+    def _create_alert_schedule_api_call(self, alert_id: str, schedule: dict, destination_id: str, warehouse_id: str):
+        """
+        Creates an alert schedule in Databricks
+        """
+        return self.client.jobs.create(
+            name=f"Alert {alert_id} schedule",
+            schedule=self._create_cron_schedule(schedule),
+            tasks=[
+                Task(
+                    task_key="alert",
+                    sql_task=SqlTask(alert=SqlTaskAlert(alert_id=alert_id, subscriptions=[
+                        SqlTaskSubscription(destination_id=destination_id)
+                    ]), warehouse_id=warehouse_id),
+                )
+            ],
+
+        )
+
+    def _create_cron_schedule(self, schedule: dict) -> CronSchedule:
+        """
+        Creates a cron schedule from Redash schedule
+        """
+        if schedule['interval']:
+            quarts_expression = self._build_quartz_expression(schedule['interval'])
+            return CronSchedule(
+                quartz_cron_expression=quarts_expression,
+                timezone_id="UTC"
+            )
+        raise ValueError("Only interval-based schedules are supported")
+
+    def _build_quartz_expression(self, interval: int) -> str:
+        """
+        Builds a quartz expression from an interval
+        """
+        if interval < 60:
+            seconds = f"*/{interval}"
+            minutes = "*"
+            hours = "*"
+        elif interval < 3600:
+            seconds = f"{interval % 60}"
+            minutes = f"*/{interval // 60}"
+            hours = '*'
+        elif interval < 86400:
+            seconds = f"{interval % 60}"
+            minutes = f"{(interval // 60) % 60}"
+            hours = f"*/{(interval // 60) // 60}"
+        else:
+            raise ValueError("Interval is too large")
+
+        return f"{seconds} {minutes} {hours} ? * * *"
+
+    def _get_path_object_id(self, path: str) -> int:
```

Comment:
@samithdisal this is the one that Shabbir helped us resolve. I feel this is an important part of the Databricks API (and undocumented). We should make it a public method in this class - since its part of the Databricks contract.

Reply:
Made this a public method and updated the doc string to document the parameters and return value.